contracts/contracts/vault/VaultCore.sol

```diff
@@ -359,31 +359,39 @@ contract VaultCore is VaultStorage {
      *      strategies and update the supply of OUSD.
      * @return uint256 New total supply of OUSD
      */
-    function rebase()
-        public
-        whenNotRebasePaused
-        nonReentrant
-        returns (uint256 newTotalSupply)
-    {
-        return _rebase();
+    function rebase() public whenNotRebasePaused nonReentrant {
+        _rebase();
     }
 
     /**
      * @dev Calculate the total value of assets held by the Vault and all
-     *      strategies and update the supply of OUSD.
+     *      strategies and update the supply of OUSD, optionaly sending a
+     *      portion of the yield to the trustee.
      * @return uint256 New total supply of OUSD
      */
-    function _rebase()
-        internal
-        whenNotRebasePaused
-        returns (uint256 newTotalSupply)
-    {
-        if (oUSD.totalSupply() == 0) return 0;
-        uint256 oldTotalSupply = oUSD.totalSupply();
-        newTotalSupply = _totalValue();
-        // Only rachet upwards
-        if (newTotalSupply > oldTotalSupply) {
-            oUSD.changeSupply(newTotalSupply);
+    function _rebase() internal whenNotRebasePaused {
+        uint256 ousdSupply = oUSD.totalSupply();
+        if (ousdSupply == 0) {
+            return;
+        }
+        uint256 vaultSupply = _totalValue();
+
+        // Yield fee collection
+        address _trusteeAddress = trusteeAddress; // gas savings
+        if (_trusteeAddress != address(0) && (vaultSupply > ousdSupply)) {
+            uint256 yield = vaultSupply.sub(ousdSupply);
+            uint256 fee = yield.mul(trusteeFeeBasis).div(10000);
+            require(yield > fee, "Fee must not be greater than yield");
```

Comment:
Yield should not be able to be zero inside this "if" statement since the "if" conditional checks that vaultSupply > ousdSupply. 

There's a test checking this behaviour. { yield: "0", basis: 1000, expectedFee: "0"},

Reply:
doh. yes you are right!